[telemetry] Adds cloud provider metadata.

lukeelmers opened the pull request:
### Summary
This PR revives some old logic from the `monitoring` plugin to detect the current cloud environment Kibana is running in (AWS, GCP, or Azure), and provide some basic stats for telemetry purposes.

### Changes
- Relocates the cloud detection logic from monitoring to `kibana_usage_collection`
- Creates a `cloud_provider` usage collector that fetches the metrics (see sample data below)
- Converts everything to TS
- I purposefully avoided rewriting all of the detector logic in this PR as the old code was still working fine after some initial testing, however there is plenty of cleanup we could still do in followup PRs. Open to any suggestions folks have here.

### Sample Data
```js
{
  name: 'aws',
  vm_type: 't2.micro',
  region: 'us-west-2',
  zone: 'us-west-2a',
}
```

### Testing
Testing is a pain because the only way to verify the collector works is by actually deploying Kibana to each of the cloud providers. Since a proper functional test which deploys to each provider isn't possible with the FTR, we've agreed that for now we'll stick with unit tests that have those APIs mocked.

Instead, I'm manually testing this works by deploying this branch to each cloud provider using [these Terraform configs](https://github.com/lukeelmers/terraform-kibana-dev). If you'd like to test as well, you can do so following the instructions in the repo, just be sure to configure your repo to be `https://github.com/lukeelmers/kibana` and your branch to be `feat/cloud-telemetry`.

To confirm everything's working, I inspect the sample telemetry payload which you can get from scrolling to the bottom of `Stack Management > Advanced Settings`. The `cloud_provider` should be provided in the payload.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/)
* Commit: 0178891cda2cfb5f74d2ed1b572b4ec763c0e168
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/execution/node/122/log)
* [Check Telemetry Schema](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/execution/node/434/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/execution/node/163/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/","number":"114482","commit":"0178891cda2cfb5f74d2ed1b572b4ec763c0e168"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/)
* Commit: 7d9ea7683ce83f6dc5851a4e155cd184c024b552
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/7d9ea7683ce83f6dc5851a4e155cd184c024b552)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #114666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/) failed 0cff9816b7cb4efa45bd6bfe20566a9522408c88
* :broken_heart: [Build #114482](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/) failed 0178891cda2cfb5f74d2ed1b572b4ec763c0e168

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/","number":"114482","commit":"0178891cda2cfb5f74d2ed1b572b4ec763c0e168"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/","number":"114666","commit":"0cff9816b7cb4efa45bd6bfe20566a9522408c88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/)
* Commit: 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Telemetry Schema](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/execution/node/374/log)

### History
* :green_heart: [Build #114701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/) succeeded 7d9ea7683ce83f6dc5851a4e155cd184c024b552
* :broken_heart: [Build #114666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/) failed 0cff9816b7cb4efa45bd6bfe20566a9522408c88
* :broken_heart: [Build #114482](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/) failed 0178891cda2cfb5f74d2ed1b572b4ec763c0e168

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/","number":"114482","commit":"0178891cda2cfb5f74d2ed1b572b4ec763c0e168"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/","number":"114666","commit":"0cff9816b7cb4efa45bd6bfe20566a9522408c88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/)
* Commit: bf99cc0b301ae219576b92bc49f7e57d5aa1102a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/execution/node/386/log)

### History
* :broken_heart: [Build #115789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/) failed 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8
* :green_heart: [Build #114701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/) succeeded 7d9ea7683ce83f6dc5851a4e155cd184c024b552
* :broken_heart: [Build #114666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/) failed 0cff9816b7cb4efa45bd6bfe20566a9522408c88
* :broken_heart: [Build #114482](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/) failed 0178891cda2cfb5f74d2ed1b572b4ec763c0e168

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/","number":"114482","commit":"0178891cda2cfb5f74d2ed1b572b4ec763c0e168"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/","number":"114666","commit":"0cff9816b7cb4efa45bd6bfe20566a9522408c88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/)
* Commit: 513377f945936e0a93296ca30df8b7650a786adf
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/513377f945936e0a93296ca30df8b7650a786adf)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #115813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/) failed bf99cc0b301ae219576b92bc49f7e57d5aa1102a
* :broken_heart: [Build #115789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/) failed 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8
* :green_heart: [Build #114701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/) succeeded 7d9ea7683ce83f6dc5851a4e155cd184c024b552
* :broken_heart: [Build #114666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/) failed 0cff9816b7cb4efa45bd6bfe20566a9522408c88
* :broken_heart: [Build #114482](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/) failed 0178891cda2cfb5f74d2ed1b572b4ec763c0e168

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114482/","number":"114482","commit":"0178891cda2cfb5f74d2ed1b572b4ec763c0e168"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/","number":"114666","commit":"0cff9816b7cb4efa45bd6bfe20566a9522408c88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/)
* Commit: 33358d331fe3bca5a030da3dbc05ce39514795eb
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/execution/node/415/log)

### History
* :green_heart: [Build #115841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/) succeeded 513377f945936e0a93296ca30df8b7650a786adf
* :broken_heart: [Build #115813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/) failed bf99cc0b301ae219576b92bc49f7e57d5aa1102a
* :broken_heart: [Build #115789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/) failed 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8
* :green_heart: [Build #114701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/) succeeded 7d9ea7683ce83f6dc5851a4e155cd184c024b552
* :broken_heart: [Build #114666](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/) failed 0cff9816b7cb4efa45bd6bfe20566a9522408c88

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114666/","number":"114666","commit":"0cff9816b7cb4efa45bd6bfe20566a9522408c88"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"}]}
PIPELINE-->

lukeelmers reviewed (commented):
It looks like Github is showing some of the files I converted to TS as new, but really I just copied the contents of the old js files & only converted to TS with minor formatting updates. The main new thing introduced in this PR is the collector itself.

There's still plenty of cleanup we _could_ do on the old code, just a question of what's worth taking the time to do now. I resisted the urge to overhaul everything, but open to any feedback folks may have 🙂

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/)
* Commit: 8adf9eb0cdfd207a564616a33d232952d2e0b627
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/8adf9eb0cdfd207a564616a33d232952d2e0b627)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #116337](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/) failed 33358d331fe3bca5a030da3dbc05ce39514795eb
* :green_heart: [Build #115841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/) succeeded 513377f945936e0a93296ca30df8b7650a786adf
* :broken_heart: [Build #115813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/) failed bf99cc0b301ae219576b92bc49f7e57d5aa1102a
* :broken_heart: [Build #115789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/) failed 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8
* :green_heart: [Build #114701](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/) succeeded 7d9ea7683ce83f6dc5851a4e155cd184c024b552

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/114701/","number":"114701","commit":"7d9ea7683ce83f6dc5851a4e155cd184c024b552"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-core (Team:Core)

lukeelmers commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/)
* Commit: 9d8a3a185208d08088014fe012a89afcecbdaa9a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/9d8a3a185208d08088014fe012a89afcecbdaa9a)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #116605](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/) succeeded 8adf9eb0cdfd207a564616a33d232952d2e0b627
* :broken_heart: [Build #116337](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/) failed 33358d331fe3bca5a030da3dbc05ce39514795eb
* :green_heart: [Build #115841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/) succeeded 513377f945936e0a93296ca30df8b7650a786adf
* :broken_heart: [Build #115813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/) failed bf99cc0b301ae219576b92bc49f7e57d5aa1102a
* :broken_heart: [Build #115789](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/) failed 9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115789/","number":"115789","commit":"9b98ac79a55dd3f35ce85084a9b2b4419d0c88e8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"}]}
PIPELINE-->

lukeelmers commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/)
* Commit: 232ea24186dde8d8330113a47904d8734a53b36b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/232ea24186dde8d8330113a47904d8734a53b36b)
* [Documentation Changes](https://kibana_95131.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #116795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/) succeeded 9d8a3a185208d08088014fe012a89afcecbdaa9a
* :green_heart: [Build #116605](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/) succeeded 8adf9eb0cdfd207a564616a33d232952d2e0b627
* :broken_heart: [Build #116337](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/) failed 33358d331fe3bca5a030da3dbc05ce39514795eb
* :green_heart: [Build #115841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/) succeeded 513377f945936e0a93296ca30df8b7650a786adf
* :broken_heart: [Build #115813](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/) failed bf99cc0b301ae219576b92bc49f7e57d5aa1102a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115813/","number":"115813","commit":"bf99cc0b301ae219576b92bc49f7e57d5aa1102a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"}]}
PIPELINE-->

pgayvallet commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/gcp.ts:
```diff
@@ -0,0 +1,135 @@
+   */
+  _extractBody(response: Response, body?: Response['body']) {
+    if (
+      response &&
+      response.statusCode === 200 &&
```
NIT: `response?.statusCode === 200`

afharo commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_service.ts:
```diff
@@ -0,0 +1,122 @@
+   * determine if Kibana is running in it and return relevant metadata.
+   */
+  checkIfService() {
+    return this._checkIfService(this._request).catch(() => this._createUnconfirmedResponse());
```
NIT: I wonder if we should handle _known_ retryable errors 🤔 

Since we're calling it only once, during start-up. Unless the APIs we use have a high-enough SLA that we don't really need to worry about those errors at all.

What do you think?

lukeelmers replied:
> Unless the APIs we use have a high-enough SLA that we don't really need to worry about those errors

I couldn't find any SLA info on the metadata APIs[0], however it should be noted that these metadata IPs are all [link-local addresses](https://en.wikipedia.org/wiki/Link-local_address): they're only accessible from the instance itself, and the APIs are on the same physical host that's running the VM, making network failures less likely (though not impossible).

Retries would certainly make this more bulletproof, however I'd prefer to treat it as an enhancement to follow up on later if you don't feel too strongly about it. WDYT?

[0] Docs for each of the APIs:
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service
https://cloud.google.com/compute/docs/storing-retrieving-metadata
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

afharo replied:
Makes sense! Thanks for the docs! We can go ahead and revisit later :)

pgayvallet reviewed (commented):
I know this was a direct re-use of existing (old) code. Feel free to ignore most of the comments if we think cleaning up the implementation is not worth the effort or should be done as a follow-up.

afharo commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_service.ts:
```diff
@@ -0,0 +1,122 @@
+    return this._checkIfService(this._request).catch(() => this._createUnconfirmedResponse());
+  }
+
+  _checkIfService(request?: Request): Promise<CloudServiceResponse> {
+    // should always be overridden by a subclass
+    return Promise.reject(new Error('not implemented'));
+  }
```
Super-NIT: Maybe we can declare this method (or the whole class) as `abstract` so we don't need this type of implementations since we won't be allowed to use it at all. What do you think?

afharo commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_services.ts:
```diff
@@ -11,7 +12,5 @@ import { GCP } from './gcp';
- * @type {Array}
  */
-export const CLOUD_SERVICES = [AWS, GCP, AZURE];
+export const CLOUD_SERVICES: [typeof AWS, typeof GCP, typeof AZURE] = [AWS, GCP, AZURE];
```
Super-NIT: `export const CLOUD_SERVICES: CloudService[] = [AWS, GCP, AZURE];`? 

Otherwise, if each one of them are not correctly extending `CloudService`, the `typeof` won't alert us about that (at least not in this line). What do you think?

lukeelmers replied:
ended up removing the list of singletons entirely as it wasn't really doing anything

afharo commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/gcp.ts:
```diff
@@ -0,0 +1,135 @@
+    const create = this._createRequestForField;
+    const allRequests = fields.map((field) => promisify(request)(create(field)));
+    return (
+      Promise.all(allRequests)
```
Q: Should we aim for the best-effort `Promise.allSettled` to get, at least, something? Or are we aiming for an all-or-nothing approach instead?

lukeelmers replied:
Good question, this was not an intentional decision as it was just how the service was implemented previously, and TBH I hadn't noticed until you pointed this out 😄 

It would probably be better to collect whatever we can, even if some of the fields fail... though I'd probably vote to do this in a follow-up step.

afharo replied:
Deal! Let's create a follow up improvement :)

afharo reviewed (commented):
Overall looks good! Thanks for such a great effort @lukeelmers! 

I'm a bit concerned with us reporting false negatives due to the APIs failing to respond during our setup lifecycle step (and not checking again). However, checking periodically doesn't really make much sense. I wonder if we could identify any safe-to-retry errors and retry the fetch only in that case.

I also added a couple of NITs. Feel free to discard them (and I'll be happy to approve) :)

What do you think?

lukeelmers commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/aws.test.ts:
```diff
@@ -132,13 +139,14 @@ describe('AWS', () => {
       const response = AWS._parseBody(body);
 
-      expect(response.getName()).toEqual(AWS.getName());
-      expect(response.isConfirmed()).toEqual(true);
-      expect(response.toJSON()).toEqual({
+      expect(response?.getName()).toEqual(AWS.getName());
+      expect(response?.isConfirmed()).toEqual(true);
+      expect(response?.toJSON()).toEqual({
```
Looks like TS doesn't infer this correctly though:

<img width="407" alt="Screen Shot 2021-04-06 at 3 09 31 PM" src="https://user-images.githubusercontent.com/1608770/113778919-37589d00-96ea-11eb-9a04-03d06dad8906.png">

(in this case the `?` is needed as the value may be null

pgayvallet replied:
You need to add the `!` in `const response = AWS._parseBody(body)!;`

lukeelmers replied:
Ah, thank you. Overlooked that in your comment above and was wondering if you knew about some TS/Jest magic that I hadn't heard of 😉

lukeelmers commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_services.ts:
```diff
@@ -11,7 +12,5 @@ import { GCP } from './gcp';
- * @type {Array}
  */
-export const CLOUD_SERVICES = [AWS, GCP, AZURE];
+export const CLOUD_SERVICES: [typeof AWS, typeof GCP, typeof AZURE] = [AWS, GCP, AZURE];
```
ended up removing the list of singletons entirely as it wasn't really doing anything

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/)
* Commit: 0ea60412807be6893670a76e6c8bd3b4d5a4f73e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/0ea60412807be6893670a76e6c8bd3b4d5a4f73e)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/execution/node/492/log)

### History
* :green_heart: [Build #117398](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/) succeeded 232ea24186dde8d8330113a47904d8734a53b36b
* :green_heart: [Build #116795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/) succeeded 9d8a3a185208d08088014fe012a89afcecbdaa9a
* :green_heart: [Build #116605](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/) succeeded 8adf9eb0cdfd207a564616a33d232952d2e0b627
* :broken_heart: [Build #116337](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/) failed 33358d331fe3bca5a030da3dbc05ce39514795eb
* :green_heart: [Build #115841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/) succeeded 513377f945936e0a93296ca30df8b7650a786adf

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/115841/","number":"115841","commit":"513377f945936e0a93296ca30df8b7650a786adf"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"}]}
PIPELINE-->

afharo commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/)
* Commit: 6d71207313dc53f7b0b32f5883a4d8bc92ce3b45
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/6d71207313dc53f7b0b32f5883a4d8bc92ce3b45)
* [Documentation Changes](https://kibana_95131.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #117703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/) failed 0ea60412807be6893670a76e6c8bd3b4d5a4f73e
* :green_heart: [Build #117398](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/) succeeded 232ea24186dde8d8330113a47904d8734a53b36b
* :green_heart: [Build #116795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/) succeeded 9d8a3a185208d08088014fe012a89afcecbdaa9a
* :green_heart: [Build #116605](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/) succeeded 8adf9eb0cdfd207a564616a33d232952d2e0b627
* :broken_heart: [Build #116337](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/) failed 33358d331fe3bca5a030da3dbc05ce39514795eb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116337/","number":"116337","commit":"33358d331fe3bca5a030da3dbc05ce39514795eb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/","number":"118178","commit":"6d71207313dc53f7b0b32f5883a4d8bc92ce3b45"}]}
PIPELINE-->

lukeelmers commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/cloud_detector.ts:
```diff
@@ -1,28 +1,40 @@
-    // Explicitly undefined. If the value is never updated, then the property will be dropped when the data is serialized.
+    // Explicitly undefined. If the value is never updated, then
+    // the property will be dropped when the data is serialized.
     this._cloudDetails = undefined;
```
Good point, TBH I don't know what this was actually doing in the first place, as `cloudDetails` is already going to be `undefined` until it is assigned anyway...

lukeelmers commented on src/plugins/kibana_usage_collection/server/collectors/cloud/cloud_provider_collector.ts:
```diff
@@ -0,0 +1,69 @@
+}
+
+export function registerCloudProviderUsageCollector(usageCollection: UsageCollectionSetup) {
+  const cloudDetector = new CloudDetector();
+  // determine the cloud service in the background
+  cloudDetector.detectCloudService();
```
Good point, I went with option 2 (adapt `isReady`)

lukeelmers commented on src/plugins/kibana_usage_collection/server/collectors/cloud/detector/aws.ts:
```diff
@@ -0,0 +1,158 @@
+      const pathsToCheck = ['/sys/hypervisor/uuid', '/sys/devices/virtual/dmi/id/product_uuid'];
+      const promises = pathsToCheck.map((path) => promisify(this._fs.readFile)(path, 'utf8'));
+
+      return Promise.all(promises).then((uuids) => {
```
ooh, thanks for pointing that out. i definitely meant to use `allSettled` instead of `all`, as it's possible for only one file to exist. updated.

lukeelmers commented:
@pgayvallet @afharo This should be ready for a final review. I'm trying to strike a balance here between getting in the functionality we want, while avoiding blowing the PR up _too_ much by refactoring the old code.

I addressed the comments that didn't require too much refactoring, but have a few enhancements I'd prefer to do as follow-ups:
1. getting rid of the constructor args which are only there for testing purposes (`request`, `fs`), and mocking those in the tests
2. getting rid of the use of the `request` lib in favor of something like `node-fetch` which we use more widely in kibana (request looks like it's mostly used in tests atm, not many other places)
3. adding some retry logic in the event of a failure
4. update GCP logic to still return values even if a request for one of the fields fails, instead of using an all-or-nothing approach

Let me know if you feel strongly that we should include any of these in the first iteration, otherwise I will create a follow-up task for us to prioritize.

I think overall (3) is my biggest concern here, however AFAICT we did the same thing (only making 1 attempt) in our previous cloud telemetry implementation. So worst case scenario this has "feature parity" with what we used to report.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/)
* Commit: 6e26141d700c77c245a828cf37c184a4950a12aa
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/6e26141d700c77c245a828cf37c184a4950a12aa)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/execution/node/190/log)

### History
* :green_heart: [Build #118178](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/) succeeded 6d71207313dc53f7b0b32f5883a4d8bc92ce3b45
* :broken_heart: [Build #117703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/) failed 0ea60412807be6893670a76e6c8bd3b4d5a4f73e
* :green_heart: [Build #117398](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/) succeeded 232ea24186dde8d8330113a47904d8734a53b36b
* :green_heart: [Build #116795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/) succeeded 9d8a3a185208d08088014fe012a89afcecbdaa9a
* :green_heart: [Build #116605](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/) succeeded 8adf9eb0cdfd207a564616a33d232952d2e0b627

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116605/","number":"116605","commit":"8adf9eb0cdfd207a564616a33d232952d2e0b627"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/","number":"118178","commit":"6d71207313dc53f7b0b32f5883a4d8bc92ce3b45"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/","number":"118433","commit":"6e26141d700c77c245a828cf37c184a4950a12aa"}]}
PIPELINE-->

afharo commented:
@elasticmachine merge upstream

afharo reviewed (approved):
The compromises seem fair to me! LGTM :)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/)
* Commit: b7d08480f812fff6a73ab982054a3fa3ff859e4a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/b7d08480f812fff6a73ab982054a3fa3ff859e4a)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/execution/node/486/log)

### History
* :broken_heart: [Build #118433](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/) failed 6e26141d700c77c245a828cf37c184a4950a12aa
* :green_heart: [Build #118178](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/) succeeded 6d71207313dc53f7b0b32f5883a4d8bc92ce3b45
* :broken_heart: [Build #117703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/) failed 0ea60412807be6893670a76e6c8bd3b4d5a4f73e
* :green_heart: [Build #117398](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/) succeeded 232ea24186dde8d8330113a47904d8734a53b36b
* :green_heart: [Build #116795](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/) succeeded 9d8a3a185208d08088014fe012a89afcecbdaa9a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/116795/","number":"116795","commit":"9d8a3a185208d08088014fe012a89afcecbdaa9a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/","number":"118178","commit":"6d71207313dc53f7b0b32f5883a4d8bc92ce3b45"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/","number":"118433","commit":"6e26141d700c77c245a828cf37c184a4950a12aa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/","number":"118532","commit":"b7d08480f812fff6a73ab982054a3fa3ff859e4a"}]}
PIPELINE-->

lukeelmers commented:
Thanks @afharo! Follow up issue created here: https://github.com/elastic/kibana/issues/96711 

I'm looking at the remaining test failure so we can get this back to green.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118560/)
* Commit: e9f66a944c4fc70e06e07f55f102b6b20d192260
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/e9f66a944c4fc70e06e07f55f102b6b20d192260)
* [Documentation Changes](https://kibana_95131.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #118532](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/) failed b7d08480f812fff6a73ab982054a3fa3ff859e4a
* :broken_heart: [Build #118433](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/) failed 6e26141d700c77c245a828cf37c184a4950a12aa
* :green_heart: [Build #118178](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/) succeeded 6d71207313dc53f7b0b32f5883a4d8bc92ce3b45
* :broken_heart: [Build #117703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/) failed 0ea60412807be6893670a76e6c8bd3b4d5a4f73e
* :green_heart: [Build #117398](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/) succeeded 232ea24186dde8d8330113a47904d8734a53b36b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117398/","number":"117398","commit":"232ea24186dde8d8330113a47904d8734a53b36b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/","number":"118178","commit":"6d71207313dc53f7b0b32f5883a4d8bc92ce3b45"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/","number":"118433","commit":"6e26141d700c77c245a828cf37c184a4950a12aa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/","number":"118532","commit":"b7d08480f812fff6a73ab982054a3fa3ff859e4a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118560/","number":"118560","commit":"e9f66a944c4fc70e06e07f55f102b6b20d192260"}]}
PIPELINE-->

pgayvallet reviewed (approved):
> Let me know if you feel strongly that we should include any of these in the first iteration, otherwise I will create a follow-up task for us to prioritize.

Na, that SGTM

lukeelmers commented:
@elasticmachine merge upstream

lukeelmers commented:
Just did one last test deployment to GCP, Azure, AWS... looks like we are good to go 🚀

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118758/)
* Commit: 8993bd10cacbc4be1815dadc789b68958b86e5c1
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-95131/8993bd10cacbc4be1815dadc789b68958b86e5c1)
* [Documentation Changes](https://kibana_95131.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :green_heart: [Build #118560](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118560/) succeeded e9f66a944c4fc70e06e07f55f102b6b20d192260
* :broken_heart: [Build #118532](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/) failed b7d08480f812fff6a73ab982054a3fa3ff859e4a
* :broken_heart: [Build #118433](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/) failed 6e26141d700c77c245a828cf37c184a4950a12aa
* :green_heart: [Build #118178](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/) succeeded 6d71207313dc53f7b0b32f5883a4d8bc92ce3b45
* :broken_heart: [Build #117703](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/) failed 0ea60412807be6893670a76e6c8bd3b4d5a4f73e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukeelmers

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/117703/","number":"117703","commit":"0ea60412807be6893670a76e6c8bd3b4d5a4f73e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118178/","number":"118178","commit":"6d71207313dc53f7b0b32f5883a4d8bc92ce3b45"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118433/","number":"118433","commit":"6e26141d700c77c245a828cf37c184a4950a12aa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118532/","number":"118532","commit":"b7d08480f812fff6a73ab982054a3fa3ff859e4a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118560/","number":"118560","commit":"e9f66a944c4fc70e06e07f55f102b6b20d192260"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/118758/","number":"118758","commit":"8993bd10cacbc4be1815dadc789b68958b86e5c1"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/96835)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/96835">](https://github.com/elastic/kibana/pull/96835) |

This backport PR will be merged automatically after passing CI.

alexfrancoeur commented:
@lukeelmers @afharo do we need to follow up with a mapping issue in the infra repo?

afharo commented:
For now, we have to :)

cc @elastic/infra-telemetry
